tests/rules/tainted_anon_type.cs

```diff
@@ -0,0 +1,15 @@
+class Asdf {
+    class ReturnObject1 {
+        public string Secret;
+    }
+
+    public void Test1() {
+        // ruleid: tainted-anon-type
+        Log(new ReturnObject1() { Secret = GetSecret() });
```

Comment:
No problem to have it here, but this test seems unrelated to the change, am I correct?

Reply:
I included the test case because it appeared that this PR might also address #7995, but it turned out that it only works for scenarios where the source and target are in the same expression.

Removed the anonymous type test case.